pkg/scheduler/algorithm/priorities/util/topologies.go

```diff
@@ -18,36 +18,91 @@ package util
 
 import (
 	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog"
 )
 
-// GetNamespacesFromPodAffinityTerm returns a set of names
+// GetNamespacesFromPodAffinityTerm returns a list of names
 // according to the namespaces indicated in podAffinityTerm.
 // If namespaces is empty it considers the given pod's namespace.
-func GetNamespacesFromPodAffinityTerm(pod *v1.Pod, podAffinityTerm *v1.PodAffinityTerm) sets.String {
-	names := sets.String{}
-	if len(podAffinityTerm.Namespaces) == 0 {
-		names.Insert(pod.Namespace)
-	} else {
-		names.Insert(podAffinityTerm.Namespaces...)
+func GetNamespacesFromPodAffinityTerm(pod *v1.Pod, podAffinityTerm *v1.PodAffinityTerm) []string {
+	if len(podAffinityTerm.Namespaces) > 0 {
+		return podAffinityTerm.Namespaces
 	}
-	return names
+	return []string{pod.Namespace}
 }
 
 // PodMatchesTermsNamespaceAndSelector returns true if the given <pod>
 // matches the namespace and selector defined by <affinityPod>`s <term>.
-func PodMatchesTermsNamespaceAndSelector(pod *v1.Pod, namespaces sets.String, selector labels.Selector) bool {
-	if !namespaces.Has(pod.Namespace) {
-		return false
+func PodMatchesTermsNamespaceAndSelector(pod *v1.Pod, namespaces []string, ps *metav1.LabelSelector) bool {
+	for _, ns := range namespaces {
+		if pod.Namespace == ns {
+			if LabelsMatchLabelSelector(labels.Set(pod.Labels), ps) {
+				return true
+			}
```

Comment:
I think you can add `else {return false}` here, otherwise we need to iterate over the remaining namespaces which is useless.

Reply:
done

Reply:
Replace with the whole if...else... block with:

```
return LabelsMatchLabelSelector(labels.Set(pod.Labels), selector)
```

Reply:
The else cannot replaced, which is for pod.Namespace != ns. I have replaced
```
if LabelsMatchLabelSelector(labels.Set(pod.Labels), selector) {
				return true
			}
```
 to 			return LabelsMatchLabelSelector(labels.Set(pod.Labels), selector)